src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/special_attributes.jl

```diff
@@ -133,18 +133,23 @@ function chern_class(v::NormalToricVariety, k::Int; check::Bool = true)
   @req k >= 0 "Chern class index must be non-negative"
   @req k <= dim(v) "Chern class index must not exceed dimension of the toric variety"
 
-  # If thus far, no non-trivial Chern classes have been computed for this toric variety, add an "empty" vector
+  # If thus far, no non-trivial Chern classes have been computed for this toric variety, add an "empty" vector and catch degenerate cases
   if !has_attribute(v, :chern_classes)
-    cs = Vector{Union{Nothing, CohomologyClass}}(nothing, dim(v) + 1)
-    cs[1] = cohomology_class(v, one(cohomology_ring(v, check = check)))
-    cs[2] = cohomology_class(v, sum(gens(cohomology_ring(v, check = check))))
+    cs = Dict{Int64, CohomologyClass}()
+    cs[0] = cohomology_class(v, one(cohomology_ring(v, check = check)))
+    cs[1] = cohomology_class(v, sum(gens(cohomology_ring(v, check = check))))
     set_attribute!(v, :chern_classes, cs)
+    if k == 0
+      return cs[0]
+    elseif k == 1
+      return cs[1]
+    end
   end
-
+  
   # Check if the Chern class in question is known
-  cs = get_attribute(v, :chern_classes)::Vector{Union{Nothing,CohomologyClass}}
-  if cs[k+1] !== nothing
-    return cs[k+1]::CohomologyClass
+  cs = get_attribute(v, :chern_classes)::Dict{Int64, CohomologyClass}
+  if haskey(cs, k)
+    return cs[k]::CohomologyClass
```

Comment:
```suggestion
    return cs[k]
```
the type assertion is no longer needed, since the dict now has a concrete value type

Reply:
Thank you!